business_objects/team_member.py

```diff
@@ -0,0 +1,74 @@
+from datetime import datetime
+from typing import Optional
+from . import general, team, user
+from ..session import session
+from ..models import TeamMember
+
+
+def get(team_id: str, id: str):
+    return (
+        session.query(TeamMember)
+        .filter(TeamMember.team_id == team_id, TeamMember.id == id)
+        .first()
+    )
+
+
+def get_by_team_and_user(team_id: str, user_id: str) -> TeamMember:
+    return (
+        session.query(TeamMember)
+        .filter(TeamMember.team_id == team_id, TeamMember.user_id == user_id)
+        .first()
+    )
+
+
+def get_all_by_team(team_id: str) -> list:
+    return session.query(TeamMember).filter(TeamMember.team_id == team_id).all()
+
+
+def get_all_by_team_count(team_id: str) -> int:
+    return session.query(TeamMember).filter(TeamMember.team_id == team_id).count()
+
+
+def create(
+    team_id: str,
+    user_id: str,
+    created_by: str,
+    created_at: Optional[datetime] = None,
+    with_commit: bool = False,
+) -> TeamMember:
+    already_exist = get_by_team_and_user(team_id=team_id, user_id=user_id)
+    if already_exist:
+        return already_exist
+
+    team_item = team.get(team_id)
+    user_item = user.get(user_id)
+    if not team_item or not user_item:
+        return None
+    if team_item.organization_id != user_item.organization_id:
+        return None
```

Comment:
why not raise? this is something that shouln't happen (assuming only users that are in the org are selectable)

Reply:
changed